### PR TITLE
Math util descriptions

### PR DESCRIPTION
**Pull request by mkrolavs**
Math util documentation

### CODE REVIEW

File: scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      m.copySign(_: Float, _: Float) --> Attributes(documentation = "Returns the first argument `magnitude` with the sign of the second argument `sign`.")
+
+      m.cos _ --> Attributes(documentation = "Returns the cosine of the `x` argument.")
+      m.cosh _ --> Attributes(documentation = "Returns the hyperbolic cosine of the `x` argument.")
+
+      m.decrementExact(_: Int) --> Attributes(documentation = "Returns the `x` argument decremented by one.")
+      m.decrementExact(_: Long) --> Attributes(documentation = "Returns the `x` argument decremented by one.")
+
+      m.exp _ --> Attributes(documentation = "Returns the Euler's number raised to the power of the `x` argument.")
+      m.expm1 _ --> Attributes(documentation = "Return the Euler's number raised to the power of the `x` argument with one subtracted from it.")
```
Comment (pzareba22) [inline scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala:39]: I'd rephrase this. The method calculates `exp(x) - 1`, but this description could also be understood as `exp(x-1)`

File: scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+      m.floor _ --> Attributes(documentation = "Returns the largest integer value less than or equal to the `x` argument.")
+      m.floorDiv(_: Int, _:Int) --> Attributes(documentation = "Returns the largest integer value less than or equal to the algebraic quotient, " +
+        "where `x` is the dividend and `y` is the divisor.")
+      m.floorDiv(_: Long, _:Long) --> Attributes(documentation = "Returns the largest integer value less than or equal to the algebraic quotient, " +
+        "where `x` is the dividend and `y` is the divisor.")
+      m.floorMod(_: Int, _:Int) --> Attributes(documentation = "Returns the largest integer value less than or equal to the algebraic quotient, " +
+        "where `x` is the dividend and `y` is the divisor.")
+      m.floorMod(_: Long, _:Long) --> Attributes(documentation = "Returns the largest integer value less than or equal to the algebraic quotient, " +
```
Comment (pzareba22) [inline scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala:48]: This looks like a modulus instead of a quotient

File: scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        "where `x` is the dividend and `y` is the divisor.")
+
+      m.getExponent(_: Double) --> Attributes(documentation = "Returns the size of unit in the last place of the `d` argument.")
+      m.getExponent(_: Float) --> Attributes(documentation = "Returns the size of unit in the last place of the `f` argument.")
```
Comment (pzareba22) [inline scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala:52]: The javadocs on the inner method say that it returns 'an unbiased exponent', why the change in the description?

File: scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      m.nextAfter(_: Float, _: Double) --> Attributes(documentation = "Returns the floating-point number adjacent to the `start` argument in the direction of the `direction` argument.")
+      m.nextDown(_: Double) --> Attributes(documentation = "Returns the floating-point value adjacent to the `d` argument in the direction of negative infinity.")
+      m.nextDown(_: Float) --> Attributes(documentation = "Returns the floating-point value adjacent to the `f` argument in the direction of negative infinity.")
+      m.nextUp(_: Double) --> Attributes(documentation = "Returns the floating-point value adjacent to the `d` argument in the direction of positive infinity.")
+      m.nextUp(_: Float) --> Attributes(documentation = "Returns the floating-point value adjacent to the `f` argument in the direction of positive infinity.")
+
+      m.pow(_: Double, _: Double) --> Attributes(documentation = "Returns the value of the `x` argument raised to the power of the the `y` argument. ")
+      m.rint _ --> Attributes(documentation = "Returns the closest floating point value that is equal to the integer for the `x` argument.")
```
Comment (pzareba22) [inline scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala:87]: maybe "... that is equal to a mathematical integer..."

**Review by pzareba22 — CHANGES_REQUESTED** [review]

**Comment by pzareba22** [comment]
Also, please fix formatting in the file

File: scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+      m.signum(_: Double) --> Attributes(documentation = "Returns the signum function of the `x` argument.")
+      m.signum(_: Float) --> Attributes(documentation = "Returns the signum function of the `x` argument.")
+      m.signum(_: Int) --> Attributes(documentation = "Returns the signum function of the `x` argument.")
+      m.signum(_: Long) --> Attributes(documentation = "Returns the signum function of the `x` argument.")
+
+      m.sin _ --> Attributes(documentation = "Returns the sine of the `x` argument.")
+      m.sinh _ --> Attributes(documentation = "Returns the hyperbolic sine of the `x` argument.")
+      m.sqrt _ --> Attributes(documentation = "Returns the square root of the `x` argument.")
+
+      m.subtractExact(_: Int, _: Int) --> Attributes(documentation = "Returns the difference of the `x` and `y` arguments.")
```
Comment (halotukozak) [inline scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala:100]: `between`? Or point out that it's x-y

File: scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+      m.subtractExact(_: Long, _: Long) --> Attributes(documentation = "Returns the difference of the `x` and `y` arguments.")
+
+      m.tan _ --> Attributes(documentation = "Returns the tangent of the `x` argument.")
+      m.tanh _ --> Attributes(documentation = "Returns the hyperbolic tangent of the `x` argument.")
+
+      m.toDegrees _ --> Attributes(documentation = "Returns an angle measured in radians `x` converted to an approximately equivalent angle measured in degrees.")
+      m.toRadians _ --> Attributes(documentation = "Returns an angle measured in degrees `x` converted to an approximately equivalent angle measured in radians.")
+
+      m.ulp(_: Double) --> Attributes(documentation = "Returns the size of an ulp (unit in the last place) of the `x` argument.")
```
Comment (halotukozak) [inline scex-core/src/main/scala/com/avsystem/scex/presentation/MathAttributes.scala:109]: I'd make it shorter, like: "Returns the size of a unit in the last place of the `x` argument." ; subsequent lines similarly

**Review by halotukozak — CHANGES_REQUESTED** [review]

**Comment by halotukozak** [comment]
LGTM
